ADAL/tests/unit/ios/ADKeychainTokenCacheTests.m

```diff
@@ -232,7 +260,67 @@ - (void)testRemoveAllForClientId
     [self verifyCacheContainsItem:item4];
 }
 
-- (void)testRemoveAllForUserIdAndClientId
+- (void)testRemoveAllForUserIdAndClientId_whenUserIdNil_shouldReturnNo
+{
+    XCTAssertTrue([self count] == 0, "Start empty.");
+    
+    ADAuthenticationError *error;
+    XCTAssertNotNil([mStore allItems:&error]);
+    ADAssertNoError;
+    
+    //add two items with the same client ID and same user ID but differnet resource
+    ADTokenCacheItem *item1 = [self adCreateCacheItem:@"eric@contoso.com"];
+    [item1 setResource:@"resource 1"];
+    [mStore addOrUpdateItem:item1 correlationId:nil error:&error];
+    ADTokenCacheItem *item2 = [self adCreateCacheItem:@"eric@contoso.com"];
+    [item2 setResource:@"resource 2"];
+    [mStore addOrUpdateItem:item2 correlationId:nil error:&error];
+    //add another two more items
+    ADTokenCacheItem *item3 = [self adCreateCacheItem:@"jack@contoso.com"];
+    [mStore addOrUpdateItem:item3 correlationId:nil error:&error];
+    ADTokenCacheItem *item4 = [self adCreateCacheItem:@"rose@contoso.com"];
+    [mStore addOrUpdateItem:item4 correlationId:nil error:&error];
+    ADAssertNoError;
```

Comment:
nit: can we try not to use ADAssertNoError going forward? It's not saving us any code and just making it more confusing for whoever reads the code. I'm perfectly fine with having XCTAssertNil(error) in each place instead of this :)

Reply:
agree :)

Reply:
done